C++/Algorithms/Graph Theory Algorithms/depth_first_search.cpp

```diff
@@ -0,0 +1,71 @@
+//this algorithms assumes that edges are bidirectional and graph is connected
+
+#include<iostream> 
+#include<vector>
+using namespace std;
+
+vector<int> adjList[100005]; // max number of node is 10005
+
+void dfs(int node,vector<bool> &vis) 
+{
+	cout<<node<<' ';
+	vis[node] = 1; //vis[node] = true, mark node as visited
+	for(int child: adjList[node])
+	{
+		//if child is not visited then only visit it
+		if(!vis[child])
+		{
+			//now make dfs on child 
+			dfs(child, vis);
+		}
+	}
+}
+
+int main() 
+{
+	#ifndef ONLINE_JUDGE
+  freopen("input.txt","r",stdin);
+  freopen("output.txt","w",stdout);
+  #endif
```

Comment:
@sumitks866 Remove these lines, rest of the code is A-OK 👍

Reply:
Done